File: examples/hello-world/gabriel-greeting-java/src/test/java/org/organicprogramming/gabriel/greeting/javaholon/api/PublicApiTest.java
package org.organicprogramming.gabriel.greeting.javaholon.api;

import greeting.v1.Greeting;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

class PublicApiTest {
    @Test
    void listLanguagesIncludesEnglish() {
        Greeting.ListLanguagesResponse response = PublicApi.listLanguages(Greeting.ListLanguagesRequest.getDefaultInstance());
        Greeting.Language english = response.getLanguagesList().stream()
                .filter(language -> language.getCode().equals("en"))
                .findFirst()
                .orElse(null);

        assertNotNull(english);
        assertEquals("English", english.getName());
        assertEquals("English", english.getNative());
    }

    @Test
    void sayHelloUsesRequestedLanguage() {
        Greeting.SayHelloResponse response = PublicApi.sayHello(Greeting.SayHelloRequest.newBuilder()
                .setName("Alice")
                .setLangCode("fr")
                .build());

        assertEquals("Bonjour Alice", response.getGreeting());
        assertEquals("French", response.getLanguage());
        assertEquals("fr", response.getLangCode());
    }

    @Test
    void sayHelloUsesLocalizedDefaultName() {
        Greeting.SayHelloResponse response = PublicApi.sayHello(Greeting.SayHelloRequest.newBuilder()
                .setLangCode("ja")
                .build());

        assertEquals("こんにちは、マリアさん", response.getGreeting());
        assertEquals("Japanese", response.getLanguage());
        assertEquals("ja", response.getLangCode());
    }

    @Test
    void sayHelloFallsBackToEnglish() {
        Greeting.SayHelloResponse response = PublicApi.sayHello(Greeting.SayHelloRequest.newBuilder()
                .setLangCode("unknown")
                .build());

        assertEquals("Hello Mary", response.getGreeting());
        assertEquals("English", response.getLanguage());
        assertEquals("en", response.getLangCode());
    }
}

File: examples/hello-world/gabriel-greeting-kotlin/src/test/kotlin/org/organicprogramming/gabriel/greeting/kotlinholon/api/PublicApiTest.kt
package org.organicprogramming.gabriel.greeting.kotlinholon.api

import greeting.v1.Greeting
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotNull

class PublicApiTest {
    @Test
    fun listLanguagesIncludesEnglish() {
        val response = PublicApi.listLanguages(Greeting.ListLanguagesRequest.getDefaultInstance())
        val english = response.languagesList.firstOrNull { it.code == "en" }

        assertNotNull(english)
        assertEquals("English", english.name)
        assertEquals("English", english.native)
    }

    @Test
    fun sayHelloUsesRequestedLanguage() {
        val response = PublicApi.sayHello(
            Greeting.SayHelloRequest.newBuilder()
                .setName("Alice")
                .setLangCode("fr")
                .build(),
        )

        assertEquals("Bonjour Alice", response.greeting)
        assertEquals("French", response.language)
        assertEquals("fr", response.langCode)
    }

    @Test
    fun sayHelloUsesLocalizedDefaultName() {
        val response = PublicApi.sayHello(
            Greeting.SayHelloRequest.newBuilder()
                .setLangCode("ja")
                .build(),
        )

        assertEquals("こんにちは、マリアさん", response.greeting)
        assertEquals("Japanese", response.language)
        assertEquals("ja", response.langCode)
    }

    @Test
    fun sayHelloFallsBackToEnglish() {
        val response = PublicApi.sayHello(
            Greeting.SayHelloRequest.newBuilder()
                .setLangCode("unknown")
                .build(),
        )

        assertEquals("Hello Mary", response.greeting)
        assertEquals("English", response.language)
        assertEquals("en", response.langCode)
    }
}
